go/fn/object_test.go

```diff
@@ -10,6 +10,89 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// processError recover the error catch by panic
+func processError(t *testing.T, expected string, err error) {
+	v := recover()
+	if v != nil {
+		switch t := v.(type) {
+		case errKubeObjectFields:
+			err = &t
+		case *errKubeObjectFields:
+			err = t
+		case errSubObjectFields:
+			err = &t
+		case *errSubObjectFields:
+			err = t
+		case errResultEnd:
+			err = &t
+		case *errResultEnd:
+			err = t
+		case ErrAttemptToTouchUpstreamIdentifier:
+			err = &t
+		case *ErrAttemptToTouchUpstreamIdentifier:
+			err = t
+		case ErrInternalAnnotation:
+			err = &t
+		case *ErrInternalAnnotation:
+			err = t
+		default:
+			panic(v)
+		}
+	}
+	assert.Equal(t, err.Error(), expected)
+}
+
+func TestWrongKRM(t *testing.T) {
```

Comment:
Write tests separately instead of using parameterized way because recover panic can only catch error once. Not sure if there is a better way to include all test cases about error messaging testing into one Test function.